pageserver/src/virtual_file/io_engine/tokio_epoll_uring_ext.rs

```diff
@@ -24,38 +24,31 @@ struct ThreadLocalState(Arc<ThreadLocalStateInner>);
 struct ThreadLocalStateInner {
     cell: tokio::sync::OnceCell<SystemHandle>,
     launch_attempts: AtomicU32,
-    weak_self: Weak<ThreadLocalStateInner>,
+    /// populated lazily through fetch_add from [`THREAD_LOCAL_STATE_ID`]
+    thread_local_state_id: once_cell::sync::OnceCell<u64>,
 }
 
 impl ThreadLocalState {
     pub fn new() -> Self {
-        Self(Arc::new_cyclic(|weak| ThreadLocalStateInner {
+        Self(Arc::new(ThreadLocalStateInner {
             cell: tokio::sync::OnceCell::default(),
             launch_attempts: AtomicU32::new(0),
-            weak_self: Weak::clone(weak),
+            // NB: don't initialize from THREAD_LOCAL_STATE_ID here,
```

Comment:
I don't understand this, `THREAD_LOCAL_STATE_ID` is an atomic counter that goes up, it's not a thread-local.

Reply:
Right, now that we're using a counter, that's not needed.